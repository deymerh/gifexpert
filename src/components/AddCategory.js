import React, { useState } from 'react';
import propTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleChange = (e) => {
    setInputValue(e.target.value)
  }
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 3) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('')
    }
  }
  return (
    <form onSubmit={handleSubmit}>
      <input
        value={inputValue}
        onChange={handleChange}
        type="text" />
    </form>
  )
}
AddCategory.propTypes = {
  setCategories: propTypes.func.isRequired
}