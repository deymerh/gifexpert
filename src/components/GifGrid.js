import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { loading, data: images } = useFetchGifs(category);

  return loading ? <p>Cargando...</p> : (
    <>
      <h3> {category} </h3>
      <div className="card-grid animate__animated animate__fadeIn">
        {
          images.map(img => {
            return (
              <GifGridItem key={img.id} {...img} />
            )
          })
        }
      </div>
    </>
  )
}
