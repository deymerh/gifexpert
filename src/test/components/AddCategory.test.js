import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el compoente <AddCategory  />', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test('Debe mostrar el conponente <AddCategory /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value } });
  });
  test('NO debe postear la información con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() { } }) // En el objeto hay una función
    expect(setCategories).not.toHaveBeenCalled();
  });
  test('debe de llamar el set categories y limpiar la caja de texto', () => {
    // wrapper.find('form').simulate('submit', { preventDefault() { } })
    const value = 'Hola Mundo';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() { } });
    // expect(setCategories).toHaveBeenCalled(); //que haya sido llamda
    // expect(setCategories).toHaveBeenCalledTimes(1); // saber cuantas veces ha sido llamada
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); //para asegurarme de que hala sido llamada como con una funcion
    expect(input.prop('value')).toBe('')
  });
});