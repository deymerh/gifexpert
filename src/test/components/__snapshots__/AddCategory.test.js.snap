// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el compoente <AddCategory  /> Debe mostrar el conponente <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
